### PR TITLE
feat: add quantile aggregate function

### PR DESCRIPTION
**Pull request by vibhatha**
This PR includes the `quantile` and `approx_quantile` functions for aggregate functions.

### CODE REVIEW

**Comment by jvanstraten** [comment]
From just the descriptions, implementations, and a few minutes of googling I'm not at all sure what these functions and their arguments actually do. Can you please clarify in the descriptions and include argument names?

**Comment by vibhatha** [comment]
@jvanstraten I updated the PR.

**Review by jvanstraten — CHANGES_REQUESTED** [review]
We discussed this function at length offline. Could you update accordingly? Right now, the description and definition just isn't good enough IMO.

**Comment by vibhatha** [comment]
@jvanstraten I updated the PR and also for the approximate type, I created a new field called `precision` and added a description. Please check that description see if it requires modifications?

**Review by jvanstraten — APPROVED** [review]
~~LGTM now!~~

**Review by jvanstraten — CHANGES_REQUESTED** [review]
Actually, correction (sorry), the same comments I gave on the other PR for precision apply here. I guess for approximate the list items should at least be monotonously increasing/decreasing if the input is sorted ascending/descending? Again, I'm also fine with just omitting its description for now since it will be a standard option.

**Comment by vibhatha** [comment]
> Actually, correction (sorry), the same comments I gave on the other PR for precision apply here. 

I updated the description of the `precision`. 

> I guess for approximate the list items should at least be monotonously increasing/decreasing if the input is sorted ascending/descending? Again, I'm also fine with just omitting its description for now since it will be a standard option.

Shall we include options about the ordering too? If so, it shouldn't be an option of `precision` but the operator itself, right?

File: extensions/functions_arithmetic.yaml [hunk 1]
```diff
@@ -1071,6 +1071,86 @@ aggregate_functions:
...
+            options: [ NEITHER, MINIMUM, MAXIMUM, BOTH ]
+            required: true
+          - name: precision
+            description: >
+              Based on required operator performance and configured optimizations
+              on saving memory bandwidth, the precision of the end result can be
+              the highest possible accuracy of an approximation. 
+
+                - EXACT: provides the highest accurate output
+                - APPROXIMATE: provides a sub-optimal output
```
Comment (jvanstraten) [inline extensions/functions_arithmetic.yaml:1118]: > I updated the description of the `precision`.

I mean here. Or maybe you forgot to push?

**Review by jvanstraten — CHANGES_REQUESTED** [review]
> Shall we include options about the ordering too? If so, it shouldn't be an option of precision but the operator itself, right?

No, because aggregate function bindings already include specialized options for sorting prior to aggregation in protobuf land.

Also needs a rebase.

**Comment by jvanstraten** [comment]
We also need to add `ordered: true` to all these implementations. I didn't realize that wasn't the default behavior. See also #312

File: extensions/functions_arithmetic.yaml (hunk 1 above)
Comment (vibhatha) [reply extensions/functions_arithmetic.yaml]: Is it okay now?

**Comment by vibhatha** [comment]
> We also need to add `ordered: true` to all these implementations. I didn't realize that wasn't the default behavior. See also #312

@jvanstraten updated.

**Review by jvanstraten — APPROVED** [review]
I think this looks good now.

**Pull request merged**